addons/mail/static/src/models/messaging_initializer/messaging_initializer.js

```diff
@@ -200,6 +203,45 @@ function factory(dependencies) {
             this.messaging.notificationGroupManager.computeGroups();
         }
 
+        /**
+         * @param {object} mailUserSettings
+         * @param {integer} mailUserSettings.id
+         * @param {boolean} mailUserSettings.is_discuss_sidebar_category_channel_open
+         * @param {boolean} mailUserSettings.is_discuss_sidebar_category_chat_open
+         */
+        _initMailUserSettings({ id, is_discuss_sidebar_category_channel_open, is_discuss_sidebar_category_chat_open }) {
+            this.messaging.update({
+                mailUserSettingsId: id,
+            });
+            this.messaging.discuss.update({
+                categoryChannel: create({
+                    autocompleteMethod: 'channel',
+                    commandAddTitleText: this.env._t("Add or join a channel"),
+                    counterComputeMethod: 'needaction',
+                    displayName: this.env._t("Channels"),
+                    hasAddCommand: true,
+                    hasViewCommand: true,
+                    isServerOpen: is_discuss_sidebar_category_channel_open,
+                    newItemPlaceholderText: this.env._t("Find or create a channel..."),
+                    serverStateKey: 'is_discuss_sidebar_category_channel_open',
+                    sortComputeMethod: 'name',
+                    supportedChannelTypes: ['channel'],
+                }),
+                categoryChat: create({
+                    autocompleteMethod: 'chat',
+                    commandAddTitleText: this.env._t("Start a conversation"),
+                    counterComputeMethod: 'unread',
+                    displayName: this.env._t("Direct Messages"),
+                    hasAddCommand: true,
+                    isServerOpen: is_discuss_sidebar_category_chat_open,
+                    newItemPlaceholderText: this.env._t("Find or start a conversation..."),
+                    serverStateKey: 'is_discuss_sidebar_category_chat_open',
+                    sortComputeMethod: 'last_action',
+                    supportedChannelTypes: ['chat'],
+                }),
+            });
+        }
```

Comment:
It would be better to create the category by default in field, and only update here the information that was received from the server.